quicksort/quick_sort.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import "fmt"
+
+// partitionArray finds the pivot index
+func partitionArray(data []int, beg int, end int, pivotIndex *int) {
```

Comment:
@singhpratyush  I'm passing the address of `pivotIndex` so it's value will be changed. Should I return it instead of using a pointer?